Test/Unit/BoltTestCase.php

```diff
@@ -3,13 +3,318 @@
+            }
+    
+            if (\is_object($classOrObject)) {
+                return static::getObjectAttribute(
+                    $classOrObject,
+                    $attributeName
+                );
+            }
+    
+            throw InvalidArgumentHelper::factory(
+                1,
+                'class name or object'
+            );
+        }
+        
+        public static function getObjectAttribute($object, $attributeName)
+        {
+            if (!\is_object($object)) {
+                throw InvalidArgumentHelper::factory(1, 'object');
+            }
+    
+            if (!\is_string($attributeName)) {
+                throw InvalidArgumentHelper::factory(2, 'string');
+            }
+    
+            if (!\preg_match('/[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/', $attributeName)) {
+                throw InvalidArgumentHelper::factory(2, 'valid attribute name');
+            }
+    
+            try {
+                $attribute = new ReflectionProperty($object, $attributeName);
+            } catch (ReflectionException $e) {
+                $reflector = new ReflectionObject($object);
+    
+                while ($reflector = $reflector->getParentClass()) {
+                    try {
+                        $attribute = $reflector->getProperty($attributeName);
+                        break;
+                    } catch (ReflectionException $e) {
+                    }
+                }
+            }
+    
+            if (isset($attribute)) {
+                if (!$attribute || $attribute->isPublic()) {
+                    return $object->$attributeName;
+                }
+    
+                $attribute->setAccessible(true);
+                $value = $attribute->getValue($object);
+                $attribute->setAccessible(false);
+    
+                return $value;
+            }
+    
+            throw new Exception(
+                \sprintf(
+                    'Attribute "%s" not found in object.',
+                    $attributeName
+                )
+            );
+        }
+        
+        public static function assertInternalType($expected, $actual, $message = '')
+        {
+            if (!\is_string($expected)) {
+                throw InvalidArgumentHelper::factory(1, 'string');
+            }
+    
+            $constraint = new IsType(
+                $expected
+            );
+    
+            static::assertThat($actual, $constraint, $message);
+        }
+        
+        /**
+         * @param string $messageRegExp
+         *
+         * @throws Exception
+         */
+        public function expectExceptionMessageRegExp($messageRegExp)
+        {
+            if (!\is_string($messageRegExp)) {
+                throw InvalidArgumentHelper::factory(1, 'string');
+            }
+    
+            $this->expectedExceptionMessageRegExp = $messageRegExp;
         }
     }
-}
\ No newline at end of file
+}
+
+if (!class_exists('\PHPUnit\Framework\Constraint\ArraySubset')) {
```

Comment:
https://github.com/sebastianbergmann/phpunit/issues/3494
the class assertArraySubset has been deprecate in PHPUnit 9, we have to define it as needed